pkg/controller/utils.go

```diff
@@ -260,3 +261,11 @@ func exponentialBackoff(ctx context.Context, backoff wait.Backoff, condition wai
 
 	return wait.ErrWaitTimeout
 }
+
+// WatcherBuilder holds various functions which add watch controls to the passed Controller.
+type WatcherBuilder []func(controller.Controller) error
```

Comment:
I'd rather name this `WatchBuilder`.
Also, can you provide a `NewWatchBuilder` function analogous to what is there for the `SchemeBuilder` / `AddToManagerBuilder`?
Finally, I'd also recommend implementing an `AddToController` function on the `WatcherBuilder` to ease adding all the watches to a controller using your `WatchBuilder`.